source/adapters/level_zero/image.cpp

```diff
@@ -36,75 +595,178 @@ urBindlessImagesUnsampledImageHandleDestroyExp(ur_context_handle_t hContext,
   std::ignore = hContext;
   std::ignore = hDevice;
   std::ignore = hImage;
-  urPrint("[UR][L0] %s function not implemented!\n", __FUNCTION__);
-  return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+
+  return UR_RESULT_SUCCESS;
 }
 
 UR_APIEXPORT ur_result_t UR_APICALL
 urBindlessImagesSampledImageHandleDestroyExp(ur_context_handle_t hContext,
                                              ur_device_handle_t hDevice,
                                              ur_exp_image_handle_t hImage) {
-  std::ignore = hContext;
-  std::ignore = hDevice;
-  std::ignore = hImage;
-  urPrint("[UR][L0] %s function not implemented!\n", __FUNCTION__);
-  return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+  // Sampled image is a combination of unsampled image and sampler.
+  urBindlessImagesUnsampledImageHandleDestroyExp(hContext, hDevice, hImage);
+
+  return UR_RESULT_SUCCESS;
```

Comment:
we do not support sampled images - shoudl this function return SUCCESS ?

Reply:
@ph0b, bindless sampler isn't supported in IGC and L0 runtime. For instance, if there is an array of sampled images used in a kernel, the case can't work correctly. It is a technical blocker at the moment. There is an on-going discussion of the issue involving @HoppeMateusz and IGC people.

@HoppeMateusz the reason I return UR_RESULT_SUCCESS here is that I'm able to make SYCL bindless sampled image tests that only use sampled image as kernel argument, e.g [sampling_1D.cpp](https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/bindless_images/sampling_1D.cpp) [sampling_2D.cpp](https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/bindless_images/sampling_2D.cpp) and [sampling_3D.cpp](https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/bindless_images/sampling_3D.cpp), working correctly on DG2, with the help of a few draft changes in IGC and L0 runtime. Basically sampler still needs a sampler_index encoded by IGC and L0 runtime saves sampler offset to a place according to sampler_index.

Return UR_RESULT_SUCCESS here also would allow us to test bindless sampler support as the above-mentioned technical blocker is in the process of being resolved.

Reply:
ok, thanks for explanation.

Wouldnt it be better to return error code from the function call urBindlessImagesUnsampledImageHandleDestroyExp() then ? Or Is it failing?